src/robot.coffee

```diff
@@ -67,6 +69,17 @@ class Robot
       @emit 'error', err
     process.on 'uncaughtException', @onUncaughtException
 
+  # Public: Adds a custom Listener with the provided matcher and callback
```

Comment:
By making this public, it does expose `Listener.matcher` as a new API. The API seems fairly well defined, but I can pull back to private if so desired.